host/archival_test.go

```diff
@@ -23,110 +23,111 @@ package host
-	s.NoError(err)
-	s.NotNil(getDomainResp)
-	s.Equal(int32(0), *getDomainResp.Configuration.WorkflowExecutionRetentionPeriodInDays)
-	s.Equal(workflow.ArchivalStatusEnabled, *getDomainResp.Configuration.ArchivalStatus)
-	s.Equal(bucketName, *getDomainResp.Configuration.ArchivalBucketName)
-
-	workflowID := "archival-workflow-id"
-	workflowType := "archival-workflow-type"
-	taskList := "archival-task-list"
+	workflowID := "archival-continueAsNew-workflow-id"
+	workflowType := "archival-continueAsNew-workflow-type"
+	taskList := "archival-continueAsNew-task-list"
 	numActivities := 1
-	runID := s.startAndFinishWorkflow(workflowID, workflowType, taskList, domain, numActivities)
-
-	// getHistoryReq without runID will cause history to attempt to be read from persistence
-	// the correct way to do this should be to update dynamic config
-	// but there is not currently a good way to do this in test the framework
-	getHistoryReq := &workflow.GetWorkflowExecutionHistoryRequest{
-		Domain: common.StringPtr(domain),
-		Execution: &workflow.WorkflowExecution{
-			WorkflowId: common.StringPtr(workflowID),
-		},
+	numRuns := 5
+	runIDs := s.startAndFinishWorkflow(workflowID, workflowType, taskList, s.archivalDomainName, numActivities, numRuns)
+
+	domainID := s.getDomainID(s.archivalDomainName)
+	for _, runID := range runIDs {
+		s.validateHistoryArchived(s.archivalDomainName, workflowID, runID)
+		s.validateHistoryDeleted(domainID, workflowID, runID)
 	}
-	s.True(s.historyDeletedFromPersistence(getHistoryReq))
-	s.False(s.historyArchived(getHistoryReq))
-	// update getHistoryReq to include runID thereby enabling history to be fetched from archival
-	getHistoryReq.Execution.RunId = common.StringPtr(runID)
-	s.True(s.historyArchived(getHistoryReq))
 }
 
-func (s *integrationSuite) historyDeletedFromPersistence(getHistoryReq *workflow.GetWorkflowExecutionHistoryRequest) bool {
-	for i := 0; i < 10; i++ {
-		_, err := s.engine.GetWorkflowExecutionHistory(createContext(), getHistoryReq)
-		if err != nil {
-			_, ok := err.(*workflow.EntityNotExistsError)
-			return ok
+func (s *integrationSuite) getDomainID(domain string) string {
+	domainResp, err := s.engine.DescribeDomain(createContext(), &workflow.DescribeDomainRequest{
+		Name: common.StringPtr(s.archivalDomainName),
+	})
+	s.Nil(err)
+	return domainResp.DomainInfo.GetUUID()
+}
+
+func (s *integrationSuite) validateHistoryArchived(domain, workflowID, runID string) {
+	var getHistoryResp *workflow.GetWorkflowExecutionHistoryResponse
+	var err error
+	for i := 0; i < retryLimit; i++ {
+		getHistoryResp, err = s.engine.GetWorkflowExecutionHistory(createContext(), &workflow.GetWorkflowExecutionHistoryRequest{
+			Domain: common.StringPtr(s.archivalDomainName),
+			Execution: &workflow.WorkflowExecution{
+				WorkflowId: common.StringPtr(workflowID),
+				RunId:      common.StringPtr(runID),
+			},
+		})
+		if getHistoryResp != nil && getHistoryResp.GetArchived() {
+			break
```

Comment:
What do you think about changing this method is ```isHistoryArchived``` and have it return a boolean. That way instead of breaking here you can simply return true. Then in the code that calls this you just do ```s.True(isHistoryArchived(..))``` that way we only need one assertion rather than three and reading a return is more clear than reading a break statement.

Reply:
Makes sense. Will update.